# Import Dependencies

In [65]:
import numpy as np
import pandas as pd

pd.set_option('display.max_columns', 500)

# Read in Data

In [79]:
long_CS= pd.read_csv("resources\SEDA_geodist_long_CS_v21.csv")

In [80]:
long_gcs= pd.read_csv("resources\SEDA_geodist_long_GCS_v21.csv")

In [81]:
long_naep= pd.read_csv("resources\SEDA_geodist_long_NAEP_v21.csv")

In [82]:
long_state= pd.read_csv("resources\SEDA_geodist_long_State_v21.csv")

In [83]:
pool_cs= pd.read_csv("resources\SEDA_geodist_pool_CS_v21.csv")

In [84]:
poolsub_cs= pd.read_csv("resources\SEDA_geodist_poolsub_CS_v21.csv")

# Columns in the data

In [85]:
print(len(long_CS.columns))
print(len(long_gcs.columns))
print(len(long_naep.columns))
print(len(long_state.columns))
print(len(pool_cs.columns))
print(len(poolsub_cs.columns))

50
50
50
50
33
44


In [91]:
columns_df= pd.DataFrame({
    "long_CS": pd.Series(long_CS.columns),
    "long_gcs": pd.Series(long_gcs.columns),
    "long_naep": pd.Series(long_naep.columns),
    "long_state": pd.Series(long_state.columns),
    "pool_cs": pd.Series(pool_cs.columns),
    "poolsub_cs": pd.Series(poolsub_cs.columns)
})

In [92]:
columns_df

long_CS    long_gcs   long_naep  long_state       pool_cs  \
0       leaidC      leaidC      leaidC      leaidC        leaidC   
1      leaname     leaname     leaname     leaname       leaname   
2         fips        fips        fips        fips          fips   
3     stateabb    stateabb    stateabb    stateabb      stateabb   
4        grade       grade       grade       grade      subgroup   
5         year        year        year        year       gap_est   
6      subject     subject     subject     subject     tot_asmts   
7   totgyb_all  totgyb_all  totgyb_all  totgyb_all     cellcount   
8       mn_all      mn_all      mn_all      mn_all      mn_asmts   
9    mn_all_se   mn_all_se      sd_all   mn_all_se     mn_avg_ol   
10      sd_all      sd_all   mn_all_se      sd_all     mn_grd_ol   
11   sd_all_se   sd_all_se   sd_all_se   sd_all_se     mn_mth_ol   
12  totgyb_asn  totgyb_asn  totgyb_asn  totgyb_asn     mn_avg_eb   
13      mn_asn      mn_asn      mn_asn      mn_asn     mn_grd_eb   
14   mn_asn_se   mn_asn_se      sd_asn   mn_asn_se     mn_mth_eb   
15      sd_asn      sd_asn   mn_asn_se      sd_asn  mn_avg_eb_se   
16   sd_asn_se   sd_asn_se   sd_asn_se   sd_asn_se  mn_grd_eb_se   
17  totgyb_blk  totgyb_blk  totgyb_blk  totgyb_blk  mn_mth_eb_se   
18      mn_blk      mn_blk      mn_blk      mn_blk  mn_avg_ol_se   
19   mn_blk_se   mn_blk_se      sd_blk   mn_blk_se  mn_grd_ol_se   
20      sd_blk      sd_blk   mn_blk_se      sd_blk  mn_mth_ol_se   
21   sd_blk_se   sd_blk_se   sd_blk_se   sd_blk_se     sd_avg_ol   
22  totgyb_fem  totgyb_fem  totgyb_fem  totgyb_fem     sd_grd_ol   
23      mn_fem      mn_fem      mn_fem      mn_fem     sd_mth_ol   
24   mn_fem_se   mn_fem_se      sd_fem   mn_fem_se     sd_avg_eb   
25      sd_fem      sd_fem   mn_fem_se      sd_fem     sd_grd_eb   
26   sd_fem_se   sd_fem_se   sd_fem_se   sd_fem_se     sd_mth_eb   
27  totgyb_hsp  totgyb_hsp  totgyb_hsp  totgyb_hsp  sd_avg_eb_se   
28      mn_hsp      mn_hsp      mn_hsp      mn_hsp  sd_grd_eb_se   
29   mn_hsp_se   mn_hsp_se      sd_hsp   mn_hsp_se  sd_mth_eb_se   
30      sd_hsp      sd_hsp   mn_hsp_se      sd_hsp  sd_avg_ol_se   
31   sd_hsp_se   sd_hsp_se   sd_hsp_se   sd_hsp_se  sd_grd_ol_se   
32  totgyb_mal  totgyb_mal  totgyb_mal  totgyb_mal  sd_mth_ol_se   
33      mn_mal      mn_mal      mn_mal      mn_mal           NaN   
34   mn_mal_se   mn_mal_se      sd_mal   mn_mal_se           NaN   
35      sd_mal      sd_mal   mn_mal_se      sd_mal           NaN   
36   sd_mal_se   sd_mal_se   sd_mal_se   sd_mal_se           NaN   
37  totgyb_wht  totgyb_wht  totgyb_wht  totgyb_wht           NaN   
38      mn_wht      mn_wht      mn_wht      mn_wht           NaN   
39   mn_wht_se   mn_wht_se      sd_wht   mn_wht_se           NaN   
40      sd_wht      sd_wht   mn_wht_se      sd_wht           NaN   
41   sd_wht_se   sd_wht_se   sd_wht_se   sd_wht_se           NaN   
42      mn_wag      mn_wag      mn_wag      mn_wag           NaN   
43   mn_wag_se   mn_wag_se   mn_wag_se   mn_wag_se           NaN   
44      mn_wbg      mn_wbg      mn_wbg      mn_wbg           NaN   
45   mn_wbg_se   mn_wbg_se   mn_wbg_se   mn_wbg_se           NaN   
46      mn_whg      mn_whg      mn_whg      mn_whg           NaN   
47   mn_whg_se   mn_whg_se   mn_whg_se   mn_whg_se           NaN   
48      mn_mfg      mn_mfg      mn_mfg      mn_mfg           NaN   
49   mn_mfg_se   mn_mfg_se   mn_mfg_se   mn_mfg_se           NaN   

          poolsub_cs  
0             leaidC  
1            leaname  
2               fips  
3           stateabb  
4           subgroup  
5            gap_est  
6      tot_asmts_ela  
7      tot_asmts_mth  
8      cellcount_ela  
9      cellcount_mth  
10      mn_asmts_ela  
11      mn_asmts_mth  
12     mn_avg_mth_ol  
13     mn_avg_ela_ol  
14     mn_grd_mth_ol  
15     mn_grd_ela_ol  
16     mn_avg_mth_eb  
17     mn_avg_ela_eb  
18     mn_grd_mth_eb  
19     mn_grd_ela_eb  
20  mn_avg_mth_eb_se  
21  mn_avg_ela_eb_se  
22  mn_gr

# Off with their heads!

In [101]:
long_CS.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
0  100002  Alabama Youth Services     1       AL      8  2009     ela   
1  100002  Alabama Youth Services     1       AL      8  2009    math   
2  100002  Alabama Youth Services     1       AL      8  2011     ela   
3  100002  Alabama Youth Services     1       AL      8  2011    math   
4  100002  Alabama Youth Services     1       AL      8  2012     ela   

   totgyb_all    mn_all  mn_all_se    sd_all  sd_all_se  totgyb_asn  mn_asn  \
0        29.0 -1.569190   0.209354  0.931462   0.016600         NaN     NaN   
1        36.0 -1.761643   0.206164  0.892374   0.021631         NaN     NaN   
2        29.0 -0.962804   0.203069  0.905344   0.020444         NaN     NaN   
3        27.0 -1.641117   0.223450  0.857457   0.018269         NaN     NaN   
4        33.0 -1.069076   0.190945  0.913058   0.014726         NaN     NaN   

   mn_asn_se  sd_asn  sd_asn_se  totgyb_blk    mn_blk  mn_blk_se    sd_blk  \
0        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
1        NaN     NaN        NaN        20.0 -1.986859   0.284851  0.823336   
2        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
3        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
4        NaN     NaN        NaN         NaN       NaN        NaN       NaN   

   sd_blk_se  totgyb_fem  mn_fem  mn_fem_se  sd_fem  sd_fem_se  totgyb_hsp  \
0        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
1   0.022021         NaN     NaN        NaN     NaN        NaN         NaN   
2        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
3        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
4        NaN         NaN     NaN        NaN     NaN        NaN         NaN   

   mn_hsp  mn_hsp_se  sd_hsp  sd_hsp_se  totgyb_mal    mn_mal  mn_mal_se  \
0     NaN        NaN     NaN        NaN        26.0 -1.631368   0.224605   
1     NaN        NaN     NaN        NaN        33.0 -1.814169   0.223662   
2     NaN        NaN     NaN        NaN        27.0 -0.990942   0.211828   
3     NaN        NaN     NaN        NaN        26.0 -1.583394   0.238398   
4     NaN        NaN     NaN        NaN        33.0 -1.134681   0.190324   

     sd_mal  sd_mal_se  totgyb_wht  mn_wht  mn_wht_se  sd_wht  sd_wht_se  \
0  0.948279   0.018422         NaN     NaN        NaN     NaN        NaN   
1  0.944506   0.024233         NaN     NaN        NaN     NaN        NaN   
2  0.912120   0.021901         NaN     NaN        NaN     NaN        NaN   
3  0.908649   0.020670         NaN     NaN        NaN     NaN        NaN   
4  0.908174   0.016259         NaN     NaN        NaN     NaN        NaN   

   mn_wag  mn_wag_se  mn_wbg  mn_wbg_se  mn_whg  mn_whg_se  mn_mfg  mn_mfg_se  
0     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
1     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
2     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
3     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
4     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN

In [102]:
long_gcs.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
0  100002  Alabama Youth Services     1       AL      8  2009     ela   
1  100002  Alabama Youth Services     1       AL      8  2009    math   
2  100002  Alabama Youth Services     1       AL      8  2011     ela   
3  100002  Alabama Youth Services     1       AL      8  2011    math   
4  100002  Alabama Youth Services     1       AL      8  2012     ela   

   totgyb_all    mn_all  mn_all_se    sd_all  sd_all_se  totgyb_asn  mn_asn  \
0        29.0  3.364652   0.618425  2.751515   0.049036         NaN     NaN   
1        36.0  2.138392   0.685979  2.969243   0.071975         NaN     NaN   
2        29.0  5.155901   0.599859  2.674363   0.060393         NaN     NaN   
3        27.0  2.539424   0.743498  2.853061   0.060789         NaN     NaN   
4        33.0  4.841975   0.564046  2.697150   0.043501         NaN     NaN   

   mn_asn_se  sd_asn  sd_asn_se  totgyb_blk    mn_blk  mn_blk_se    sd_blk  \
0        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
1        NaN     NaN        NaN        20.0  1.389016   0.947799  2.739531   
2        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
3        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
4        NaN     NaN        NaN         NaN       NaN        NaN       NaN   

   sd_blk_se  totgyb_fem  mn_fem  mn_fem_se  sd_fem  sd_fem_se  totgyb_hsp  \
0        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
1   0.073273         NaN     NaN        NaN     NaN        NaN         NaN   
2        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
3        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
4        NaN         NaN     NaN        NaN     NaN        NaN         NaN   

   mn_hsp  mn_hsp_se  sd_hsp  sd_hsp_se  totgyb_mal    mn_mal  mn_mal_se  \
0     NaN        NaN     NaN        NaN        26.0  3.180978   0.663477   
1     NaN        NaN     NaN        NaN        33.0  1.963618   0.744201   
2     NaN        NaN     NaN        NaN        27.0  5.072781   0.625734   
3     NaN        NaN     NaN        NaN        26.0  2.731487   0.793233   
4     NaN        NaN     NaN        NaN        33.0  4.648181   0.562213   

     sd_mal  sd_mal_se  totgyb_wht  mn_wht  mn_wht_se  sd_wht  sd_wht_se  \
0  2.801194   0.054420         NaN     NaN        NaN     NaN        NaN   
1  3.142704   0.080630         NaN     NaN        NaN     NaN        NaN   
2  2.694380   0.064694         NaN     NaN        NaN     NaN        NaN   
3  3.023397   0.068775         NaN     NaN        NaN     NaN        NaN   
4  2.682725   0.048029         NaN     NaN        NaN     NaN        NaN   

   mn_wag  mn_wag_se  mn_wbg  mn_wbg_se  mn_whg  mn_whg_se  mn_mfg  mn_mfg_se  
0     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
1     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
2     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
3     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
4     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN

In [103]:
long_naep.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
0  100002  Alabama Youth Services     1       AL      8  2009     ela   
1  100002  Alabama Youth Services     1       AL      8  2009    math   
2  100002  Alabama Youth Services     1       AL      8  2011     ela   
3  100002  Alabama Youth Services     1       AL      8  2011    math   
4  100002  Alabama Youth Services     1       AL      8  2012     ela   

   totgyb_all     mn_all     sd_all  mn_all_se  sd_all_se  totgyb_asn  mn_asn  \
0        29.0  209.40759  32.880596   7.390182   0.585985         NaN     NaN   
1        36.0  217.34308  33.107059   7.648670   0.802523         NaN     NaN   
2        29.0  230.81302  31.958628   7.168319   0.721690         NaN     NaN   
3        27.0  221.81458  31.811636   8.290004   0.677796         NaN     NaN   
4        33.0  227.06160  32.230938   6.740346   0.519832         NaN     NaN   

   sd_asn  mn_asn_se  sd_asn_se  totgyb_blk     mn_blk     sd_blk  mn_blk_se  \
0     NaN        NaN        NaN         NaN        NaN        NaN        NaN   
1     NaN        NaN        NaN        20.0  208.98753  30.545776  10.567958   
2     NaN        NaN        NaN         NaN        NaN        NaN        NaN   
3     NaN        NaN        NaN         NaN        NaN        NaN        NaN   
4     NaN        NaN        NaN         NaN        NaN        NaN        NaN   

   sd_blk_se  totgyb_fem  mn_fem  sd_fem  mn_fem_se  sd_fem_se  totgyb_hsp  \
0        NaN         NaN     NaN     NaN        NaN        NaN         NaN   
1   0.816991         NaN     NaN     NaN        NaN        NaN         NaN   
2        NaN         NaN     NaN     NaN        NaN        NaN         NaN   
3        NaN         NaN     NaN     NaN        NaN        NaN         NaN   
4        NaN         NaN     NaN     NaN        NaN        NaN         NaN   

   mn_hsp  sd_hsp  mn_hsp_se  sd_hsp_se  totgyb_mal     mn_mal     sd_mal  \
0     NaN     NaN        NaN        NaN        26.0  207.21269  33.474255   
1     NaN     NaN        NaN        NaN        33.0  215.39435  35.041161   
2     NaN     NaN        NaN        NaN        27.0  229.81973  32.197830   
3     NaN     NaN        NaN        NaN        26.0  223.95609  33.710880   
4     NaN     NaN        NaN        NaN        33.0  224.74576  32.058556   

   mn_mal_se  sd_mal_se  totgyb_wht  mn_wht  sd_wht  mn_wht_se  sd_wht_se  \
0   7.928550   0.650314         NaN     NaN     NaN        NaN        NaN   
1   8.297847   0.899029         NaN     NaN     NaN        NaN        NaN   
2   7.477520   0.773098         NaN     NaN     NaN        NaN        NaN   
3   8.844554   0.766845         NaN     NaN     NaN        NaN        NaN   
4   6.718442   0.573944         NaN     NaN     NaN        NaN        NaN   

   mn_wag  mn_wag_se  mn_wbg  mn_wbg_se  mn_whg  mn_whg_se  mn_mfg  mn_mfg_se  
0     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
1     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
2     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
3     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
4     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN

In [104]:
long_state.head()

leaidC                 leaname  fips stateabb  grade  year subject  \
0  100002  Alabama Youth Services     1       AL      8  2009     ela   
1  100002  Alabama Youth Services     1       AL      8  2009    math   
2  100002  Alabama Youth Services     1       AL      8  2011     ela   
3  100002  Alabama Youth Services     1       AL      8  2011    math   
4  100002  Alabama Youth Services     1       AL      8  2012     ela   

   totgyb_all    mn_all  mn_all_se    sd_all  sd_all_se  totgyb_asn  mn_asn  \
0        29.0 -1.168019   0.190559  0.920672   0.005181         NaN     NaN   
1        36.0 -1.282802   0.190213  0.887760   0.005257         NaN     NaN   
2        29.0 -0.708781   0.188333  0.920245   0.005173         NaN     NaN   
3        27.0 -1.233343   0.215231  0.885889   0.005659         NaN     NaN   
4        33.0 -0.804737   0.179870  0.928212   0.005347         NaN     NaN   

   mn_asn_se  sd_asn  sd_asn_se  totgyb_blk    mn_blk  mn_blk_se    sd_blk  \
0        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
1        NaN     NaN        NaN        20.0 -1.496076   0.265494  0.825772   
2        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
3        NaN     NaN        NaN         NaN       NaN        NaN       NaN   
4        NaN     NaN        NaN         NaN       NaN        NaN       NaN   

   sd_blk_se  totgyb_fem  mn_fem  mn_fem_se  sd_fem  sd_fem_se  totgyb_hsp  \
0        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
1    0.00961         NaN     NaN        NaN     NaN        NaN         NaN   
2        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
3        NaN         NaN     NaN        NaN     NaN        NaN         NaN   
4        NaN         NaN     NaN        NaN     NaN        NaN         NaN   

   mn_hsp  mn_hsp_se  sd_hsp  sd_hsp_se  totgyb_mal    mn_mal  mn_mal_se  \
0     NaN        NaN     NaN        NaN        26.0 -1.225511   0.204771   
1     NaN        NaN     NaN        NaN        33.0 -1.332543   0.207005   
2     NaN        NaN     NaN        NaN        27.0 -0.735563   0.196830   
3     NaN        NaN     NaN        NaN        26.0 -1.176524   0.230360   
4     NaN        NaN     NaN        NaN        33.0 -0.867299   0.179208   

     sd_mal  sd_mal_se  totgyb_wht  mn_wht  mn_wht_se  sd_wht  sd_wht_se  \
0  0.935234   0.008267         NaN     NaN        NaN     NaN        NaN   
1  0.934875   0.009110         NaN     NaN        NaN     NaN        NaN   
2  0.926287   0.008442         NaN     NaN        NaN     NaN        NaN   
3  0.934023   0.009104         NaN     NaN        NaN     NaN        NaN   
4  0.923844   0.008249         NaN     NaN        NaN     NaN        NaN   

   mn_wag  mn_wag_se  mn_wbg  mn_wbg_se  mn_whg  mn_whg_se  mn_mfg  mn_mfg_se  
0     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
1     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
2     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
3     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN  
4     NaN        NaN     NaN        NaN     NaN        NaN     NaN        NaN

In [105]:
pool_cs.head()

leaidC                 leaname  fips stateabb subgroup  gap_est  tot_asmts  \
0  100002  Alabama Youth Services     1       AL      all        0        185   
1  100002  Alabama Youth Services     1       AL      blk        0         20   
2  100002  Alabama Youth Services     1       AL      mal        0        176   
3  100005        Albertville City     1       AL      all        0      26185   
4  100005        Albertville City     1       AL      fem        0      11421   

   cellcount    mn_asmts  mn_avg_ol  mn_grd_ol  mn_mth_ol  mn_avg_eb  \
0          6   30.833334        NaN        NaN        NaN        NaN   
1          1   20.000000        NaN        NaN        NaN        NaN   
2          6   29.333334        NaN        NaN        NaN        NaN   
3         82  319.329250  -0.280656  -0.061207  -0.116414  -0.280273   
4         69  165.521740  -0.225336  -0.040484  -0.190132  -0.225179   

   mn_grd_eb  mn_mth_eb  mn_avg_eb_se  mn_grd_eb_se  mn_mth_eb_se  \
0        NaN        NaN           NaN           NaN           NaN   
1        NaN        NaN           NaN           NaN           NaN   
2        NaN        NaN           NaN           NaN           NaN   
3  -0.058287  -0.112834      0.014924      0.011383      0.029116   
4  -0.037043  -0.187486      0.016971      0.012768      0.032495   

   mn_avg_ol_se  mn_grd_ol_se  mn_mth_ol_se  sd_avg_ol  sd_grd_ol  sd_mth_ol  \
0           NaN           NaN           NaN        NaN        NaN        NaN   
1           NaN           NaN           NaN        NaN        NaN        NaN   
2           NaN           NaN           NaN        NaN        NaN        NaN   
3      0.014940      0.011783      0.029962   0.845352  -0.010274  -0.019646   
4      0.017021      0.013442      0.033580   0.768277        NaN  -0.020952   

   sd_avg_eb  sd_grd_eb  sd_mth_eb  sd_avg_eb_se  sd_grd_eb_se  sd_mth_eb_se  \
0        NaN        NaN        NaN           NaN           NaN           NaN   
1        NaN        NaN        NaN           NaN           NaN           NaN   
2        NaN        NaN        NaN           NaN           NaN           NaN   
3   0.845379  -0.008705  -0.021662      0.007639      0.005382      0.014190   
4   0.772706        NaN  -0.026816      0.010658           NaN      0.017749   

   sd_avg_ol_se  sd_grd_ol_se  sd_mth_ol_se  
0           NaN           NaN           NaN  
1           NaN           NaN           NaN  
2           NaN           NaN           NaN  
3      0.007737      0.006004      0.015404  
4      0.011180           NaN      0.021319

In [106]:
poolsub_cs.head()

leaidC                 leaname  fips stateabb subgroup  gap_est  \
0  100002  Alabama Youth Services     1       AL      all        0   
1  100002  Alabama Youth Services     1       AL      blk        0   
2  100002  Alabama Youth Services     1       AL      mal        0   
3  100005        Albertville City     1       AL      all        0   
4  100005        Albertville City     1       AL      fem        0   

   tot_asmts_ela  tot_asmts_mth  cellcount_ela  cellcount_mth  mn_asmts_ela  \
0           91.0           94.0            3.0            3.0     30.333334   
1            NaN           20.0            NaN            1.0           NaN   
2           86.0           90.0            3.0            3.0     28.666666   
3        12754.0        13431.0           40.0           42.0    318.850010   
4         5621.0         5800.0           34.0           35.0    165.323530   

   mn_asmts_mth  mn_avg_mth_ol  mn_avg_ela_ol  mn_grd_mth_ol  mn_grd_ela_ol  \
0     31.333334            NaN            NaN            NaN            NaN   
1     20.000000            NaN            NaN            NaN            NaN   
2     30.000000            NaN            NaN            NaN            NaN   
3    319.785710      -0.339745      -0.222717      -0.060150      -0.061749   
4    165.714280      -0.323230      -0.129925      -0.032657      -0.048354   

   mn_avg_mth_eb  mn_avg_ela_eb  mn_grd_mth_eb  mn_grd_ela_eb  \
0            NaN            NaN            NaN            NaN   
1            NaN            NaN            NaN            NaN   
2            NaN            NaN            NaN            NaN   
3      -0.337170      -0.223722      -0.061004      -0.054038   
4      -0.319693      -0.132169      -0.036577      -0.034901   

   mn_avg_mth_eb_se  mn_avg_ela_eb_se  mn_grd_mth_eb_se  mn_grd_ela_eb_se  \
0               NaN               NaN               NaN               NaN   
1               NaN               NaN               NaN               NaN   
2               NaN               NaN               NaN               NaN   
3          0.020049          0.020477          0.014545          0.014061   
4          0.022668          0.023309          0.016292          0.015155   

   mn_avg_mth_ol_se  mn_grd_mth_ol_se  mn_avg_ela_ol_se  mn_grd_ela_ol_se  \
0               NaN               NaN               NaN               NaN   
1               NaN               NaN               NaN               NaN   
2               NaN               NaN               NaN               NaN   
3          0.020326          0.015804          0.020785          0.016567   
4          0.023162          0.018212          0.023779          0.018764   

   sd_avg_mth_ol  sd_avg_ela_ol  sd_grd_mth_ol  sd_grd_ela_ol  sd_avg_mth_eb  \
0            NaN            NaN            NaN            NaN            NaN   
1            NaN            NaN            NaN            NaN            NaN   
2            NaN            NaN            NaN            NaN            NaN   
3       0.846394       0.858098      -0.030082            NaN       0.842975   
4       0.770740       0.775054            NaN            NaN       0.764392   

   sd_avg_ela_eb  sd_grd_mth_eb  sd_grd_ela_eb  sd_avg_mth_eb_se  \
0            NaN            NaN            NaN               NaN   
1            NaN            NaN            NaN               NaN   
2            NaN            NaN            NaN               NaN   
3       0.858925      -0.021912            NaN          0.010473   
4       0.785076            NaN            NaN          0.013753   

   sd_avg_ela_eb_se  sd_grd_mth_eb_se  sd_grd_ela_eb_se  sd_avg_mth_ol_se  \
0               NaN               NaN               NaN               NaN   
1               NaN               NaN               NaN               NaN   
2               NaN               NaN               NaN               NaN   
3          0.010814           0.00755               NaN          0.010886   
4          0.014301               NaN   

# Describe the datasets

In [94]:
desc_long_CS= long_CS.describe()
desc_long_gcs= long_gcs.describe()
desc_long_naep= long_naep.describe()
desc_long_state= long_state.describe()
desc_pool_cs= pool_cs.describe()
desc_poolsub_cs= poolsub_cs.describe()

In [95]:
desc_long_CS

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

          totgyb_all         mn_all      mn_all_se         sd_all  \
count  825814.000000  825814.000000  825814.000000  825814.000000   
mean      328.138101       0.045220       0.112220       0.876175   
std      1166.280140       0.400055       0.054619       0.121183   
min        20.000000      -3.186861       0.010884       0.042187   
25%        55.000000      -0.205949       0.070441       0.807569   
50%       119.000000       0.049993       0.099618       0.873354   
75%       277.000000       0.300706       0.149012       0.940343   
max     76800.000000       3.804560       1.841295       4.186017   

           sd_all_se    totgyb_asn        mn_asn     mn_asn_se        sd_asn  \
count  825814.000000  85523.000000  85523.000000  85523.000000  85523.000000   
mean        0.069390    122.689382      0.608329      0.190778      0.987328   
std         0.044821    483.938421      0.556105      0.085566      0.158555   
min         0.007450     20.000000     -3.400659      0.015841      0.177361   
25%         0.029866     29.000000      0.254931      0.131710      0.904609   
50%         0.066962     47.000000      0.610957      0.190276      0.975054   
75%         0.097356     97.000000      0.965965      0.235567      1.049185   
max         1.416418  12214.000000      4.290233      2.123441      3.989972   

          sd_asn_se     totgyb_blk         mn_blk      mn_blk_se  \
count  85523.000000  183171.000000  183171.000000  183171.000000   
mean       0.084771     224.051460      -0.438864       0.132186   
std        0.077507     719.344149       0.326138       0.058851   
min        0.011620      20.000000      -2.802191       0.015411   
25%        0.029321      36.000000      -0.654225       0.083788   
50%        0.053963      72.000000      -0.444535       0.125113   
75%        0.121298     170.000000      -0.225432       0.177508   
max        1.861681   22678.000000       1.476525       1.171951   

              sd_blk      sd_blk_se     totgyb_fem         mn_fem  \
count  183171.000000  183171.000000  684574.000000  684574.000000   
mean        0.849140       0.065292     188.532721       0.120130   
std         0.110419       0.049079     621.838991       0.403455   
min         0.103305       0.009407      20.000000      -2.405667   
25%         0.791534       0.021645      39.000000      -0.140214   
50%         0.844292       0.057352      74.000000       0.121420   
75%         0.906163       0.096868     159.000000       0.379138   
max         2.570137       1.111087   37799.000000       3.931363   

           mn_fem_se         sd_fem      sd_fem_se     totgyb_hsp  \
count  684574.000000  684574.000000  684574.000000  222124.000000   
mean        0.130864       0.855555       0.067869     252.636262   
std         0.055945       0.115374       0.049798    1028.612933   
min         0.014748       0.105092       0.007860      20.000000   
25%         0.087132       0.796186       0.018935      33.000000   
50%         0.122946       0.850554       0.065888      66.000000   
75%         0.171063       0.911051       0.101516     177.000000   
max         1.691966       3.085215       1.206432   38791.000000   

              mn_hsp      mn_hsp_se         sd_hsp      sd_hsp_se  \
count  222124.000000  222124.000000  222124.000000  222124.000000   
mean       -0.278085       0.135376       0.860349       0.060

In [96]:
desc_long_gcs

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

          totgyb_all         mn_all      mn_all_se         sd_all  \
count  825814.000000  825814.000000  825814.000000  825814.000000   
mean      328.138101       5.565391       0.336394       2.628670   
std      1166.280140       1.988355       0.165724       0.413478   
min        20.000000      -4.601914       0.032604       0.140370   
25%        55.000000       4.086436       0.209938       2.364256   
50%       119.000000       5.516290       0.297787       2.632125   
75%       277.000000       7.023564       0.442947       2.871308   
max     76800.000000      16.319139       5.515627      12.365393   

           sd_all_se    totgyb_asn        mn_asn     mn_asn_se        sd_asn  \
count  825814.000000  85523.000000  85523.000000  85523.000000  85523.000000   
mean        0.208084    122.689382      7.200015      0.569769      2.953043   
std         0.135414    483.938421      2.442277      0.256700      0.515028   
min         0.022315     20.000000     -5.865470      0.047452      0.523919   
25%         0.090063     29.000000      5.448634      0.393414      2.646907   
50%         0.199901     47.000000      7.109251      0.566381      2.931653   
75%         0.291724     97.000000      8.867107      0.704836      3.185928   
max         3.984298  12214.000000     22.275122      6.696728     13.276047   

          sd_asn_se     totgyb_blk         mn_blk      mn_blk_se  \
count  85523.000000  183171.000000  183171.000000  183171.000000   
mean       0.252941     224.051460       4.132336       0.396487   
std        0.231137     719.344149       1.788857       0.179498   
min        0.034808      20.000000      -3.348936       0.046091   
25%        0.087202      36.000000       2.760144       0.249794   
50%        0.161543      72.000000       4.094108       0.374391   
75%        0.362594     170.000000       5.464153       0.531483   
max        6.194471   22678.000000      12.196367       3.712929   

              sd_blk      sd_blk_se     totgyb_fem         mn_fem  \
count  183171.000000  183171.000000  684574.000000  684574.000000   
mean        2.546893       0.195934     188.532721       5.806853   
std         0.376843       0.148733     621.838991       2.037545   
min         0.276098       0.028026      20.000000      -3.786019   
25%         2.312592       0.064641      39.000000       4.270745   
50%         2.548520       0.170269      74.000000       5.754171   
75%         2.761659       0.290041     159.000000       7.313653   
max         7.592120       3.328277   37799.000000      19.613150   

           mn_fem_se         sd_fem      sd_fem_se     totgyb_hsp  \
count  684574.000000  684574.000000  684574.000000  222124.000000   
mean        0.392497       2.566808       0.203577     252.636262   
std         0.170557       0.394003       0.150433    1028.612933   
min         0.043832       0.318354       0.023545      20.000000   
25%         0.259561       2.319130       0.056467      33.000000   
50%         0.369074       2.568287       0.196127      66.000000   
75%         0.512901       2.786792       0.304254     177.000000   
max         5.161969       9.241808       3.822171   38791.000000   

              mn_hsp      mn_hsp_se         sd_hsp      sd_hsp_se  \
count  222124.000000  222124.000000  222124.000000  222124.000000   
mean        4.514001       0.405153       2.573891       0.180

In [97]:
desc_long_naep

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

          totgyb_all         mn_all         sd_all      mn_all_se  \
count  825814.000000  825814.000000  825814.000000  825814.000000   
mean      328.138101     245.368960      30.441846       3.896056   
std      1166.280140      24.817281       5.184859       1.943400   
min        20.000000     114.207150       1.565124       0.363533   
25%        55.000000     228.196378      26.897011       2.418443   
50%       119.000000     245.202560      30.546568       3.439700   
75%       277.000000     262.570257      33.708389       5.099976   
max     76800.000000     375.458440     147.766400      61.499256   

           sd_all_se    totgyb_asn        mn_asn        sd_asn     mn_asn_se  \
count  825814.000000  85523.000000  85523.000000  85523.000000  85523.000000   
mean        2.408928    122.689382    264.213928     34.154915      6.580929   
std         1.575808    483.938421     30.265167      6.139830      2.941361   
min         0.248818     20.000000     99.107666      6.260830      0.529085   
25%         1.045357     29.000000    243.636095     30.374425      4.555845   
50%         2.302719     47.000000    263.100400     34.027737      6.535615   
75%         3.374068     97.000000    283.998410     37.234781      8.154646   
max        44.424931  12214.000000    441.867650    148.027940     74.957466   

          sd_asn_se     totgyb_blk         mn_blk         sd_blk  \
count  85523.000000  183171.000000  183171.000000  183171.000000   
mean       2.920028     224.051460     228.758834      29.498467   
std        2.652629     719.344149      22.675072       4.759979   
min        0.388104      20.000000     129.180240       3.078496   
25%        1.009118      36.000000     213.230470      26.277265   
50%        1.852622      72.000000     229.219020      29.588207   
75%        4.200429     170.000000     244.799475      32.436369   
max       69.068367   22678.000000     314.946560      90.725815   

           mn_blk_se      sd_blk_se     totgyb_fem         mn_fem  \
count  183171.000000  183171.000000  684574.000000  684574.000000   
mean        4.592179       2.268489     188.532721     248.236700   
std         2.103669       1.726871     621.838991      24.501244   
min         0.513911       0.312495      20.000000     123.957090   
25%         2.881899       0.750063      39.000000     230.850810   
50%         4.330511       1.956334      74.000000     247.870575   
75%         6.143965       3.353889     159.000000     265.586045   
max        41.399170      37.110298   37799.000000     403.577090   

              sd_fem      mn_fem_se      sd_fem_se     totgyb_hsp  \
count  684574.000000  684574.000000  684574.000000  222124.000000   
mean       29.727471       4.546764       2.357524     252.636262   
std         4.981161       2.008899       1.750996    1028.612933   
min         3.804324       0.488723       0.262531      20.000000   
25%        26.303948       2.984426       0.658222      33.000000   
50%        29.795918       4.273438       2.253879      66.000000   
75%        32.748504       5.934157       3.520308     177.000000   
max       103.046170      59.726406      42.617218   38791.000000   

              mn_hsp         sd_hsp      mn_hsp_se      sd_hsp_se  \
count  222124.000000  222124.000000  222124.000000  222124.000000   
mean      233.040101      29.822799       4.694741       2.091

In [98]:
desc_long_state

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

          totgyb_all         mn_all      mn_all_se         sd_all  \
count  825814.000000  825814.000000  825814.000000  825814.000000   
mean      328.138101       0.002542       0.104061       0.903563   
std      1166.280140       0.340808       0.055142       0.104510   
min        20.000000      -2.972786       0.000000       0.251772   
25%        55.000000      -0.207796       0.062064       0.849491   
50%       119.000000       0.001459       0.092435       0.901879   
75%       277.000000       0.212722       0.142700       0.954235   
max     76800.000000       3.656457       1.760634       3.719599   

           sd_all_se    totgyb_asn        mn_asn     mn_asn_se        sd_asn  \
count  825814.000000  85523.000000  85523.000000  85523.000000  85523.000000   
mean        0.058912    122.689382      0.546172      0.179057      0.991131   
std         0.043337    483.938421      0.514938      0.082941      0.143650   
min         0.000000     20.000000     -3.424887      0.009375      0.296857   
25%         0.021279     29.000000      0.231252      0.121399      0.918437   
50%         0.058466     47.000000      0.555609      0.179640      0.971996   
75%         0.086205     97.000000      0.874363      0.223434      1.045130   
max         1.343618  12214.000000      4.114595      1.951427      3.776745   

          sd_asn_se     totgyb_blk         mn_blk      mn_blk_se  \
count  85523.000000  183171.000000  183171.000000  183171.000000   
mean       0.073129     224.051460      -0.416928       0.122927   
std        0.072831     719.344149       0.293577       0.058248   
min        0.000000      20.000000      -2.543317       0.006473   
25%        0.020539      36.000000      -0.605537       0.075641   
50%        0.046141      72.000000      -0.421108       0.116818   
75%        0.107019     170.000000      -0.230172       0.168060   
max        1.751017   22678.000000       1.738854       1.169030   

              sd_blk      sd_blk_se     totgyb_fem         mn_fem  \
count  183171.000000  183171.000000  684574.000000  684574.000000   
mean        0.876999       0.054904     188.532721       0.074448   
std         0.095191       0.046300     621.838991       0.346849   
min         0.273550       0.000000      20.000000      -2.267843   
25%         0.828387       0.011884      39.000000      -0.146036   
50%         0.875378       0.049883      74.000000       0.072006   
75%         0.921897       0.085082     159.000000       0.292048   
max         2.518226       1.123573   37799.000000       3.609545   

           mn_fem_se         sd_fem      sd_fem_se     totgyb_hsp  \
count  684574.000000  684574.000000  684574.000000  222124.000000   
mean        0.122631       0.884136       0.056920     252.636262   
std         0.055918       0.100978       0.047837    1028.612933   
min         0.005450       0.241442       0.000000      20.000000   
25%         0.079300       0.834420       0.007461      33.000000   
50%         0.115517       0.881488       0.057254      66.000000   
75%         0.163121       0.929520       0.089491     177.000000   
max         1.605439       3.055364       1.159092   38791.000000   

              mn_hsp      mn_hsp_se         sd_hsp      sd_hsp_se  \
count  222124.000000  222124.000000  222124.000000  222124.000000   
mean       -0.269196       0.125265       0.878575       0.049

In [99]:
desc_pool_cs

leaidC          fips       gap_est     tot_asmts     cellcount  \
count  7.214700e+04  72147.000000  72147.000000  7.214700e+04  72147.000000   
mean   2.932015e+06     29.187506      0.257585  1.866353e+04     62.831511   
std    1.520158e+06     15.201987      0.437307  6.298952e+04     24.394492   
min    1.000020e+05      1.000000      0.000000  2.000000e+01      1.000000   
25%    1.734045e+06     17.000000      0.000000  2.408000e+03     56.000000   
50%    2.926850e+06     29.000000      0.000000  6.301000e+03     72.000000   
75%    4.108770e+06     41.000000      1.000000  1.622100e+04     82.000000   
max    5.606240e+06     56.000000      1.000000  4.685211e+06     84.000000   

           mn_asmts     mn_avg_ol     mn_grd_ol     mn_mth_ol     mn_avg_eb  \
count  72147.000000  67235.000000  49956.000000  60872.000000  67242.000000   
mean     269.174043      0.049678     -0.006932      0.017771      0.049536   
std      907.378454      0.369447      0.046488      0.167213      0.364393   
min       20.000000     -1.841032     -0.389979     -0.749475     -1.690200   
25%       41.000000     -0.172902     -0.035998     -0.100292     -0.168033   
50%       92.738098      0.007239     -0.006866      0.013402      0.003202   
75%      235.928570      0.281770      0.022449      0.142542      0.280777   
max    70988.047000      1.907693      0.337531      0.839080      1.755396   

          mn_grd_eb     mn_mth_eb  mn_avg_eb_se  mn_grd_eb_se  mn_mth_eb_se  \
count  49956.000000  60872.000000  67242.000000  49956.000000  60872.000000   
mean      -0.006929      0.017796      0.025079      0.014581      0.037474   
std        0.040103      0.156233      0.014751      0.003376      0.009806   
min       -0.279810     -0.679975      0.004379      0.004757      0.008545   
25%       -0.032037     -0.093153      0.016375      0.011994      0.030515   
50%       -0.006770      0.011162      0.020405      0.013999      0.035813   
75%        0.018468      0.138895      0.027379      0.016665      0.043077   
max        0.226243      0.745145      0.131102      0.024750      0.071296   

       mn_avg_ol_se  mn_grd_ol_se  mn_mth_ol_se     sd_avg_ol     sd_grd_ol  \
count  67235.000000  49956.000000  60872.000000  50458.000000  37565.000000   
mean       0.028555      0.016093      0.041431      0.865058     -0.003462   
std        0.025651      0.004545      0.013068      0.063459      0.013257   
min        0.004398      0.004841      0.008633      0.625352     -0.095699   
25%        0.016478      0.012583      0.031945      0.825000     -0.011145   
50%        0.020713      0.015057      0.038357      0.857771     -0.004035   
75%        0.028598      0.018717      0.048406      0.897884      0.004283   
max        0.225080      0.029539      0.082737      1.546541      0.076853   

          sd_mth_ol     sd_avg_eb     sd_grd_eb     sd_mth_eb  sd_avg_eb_se  \
count  46960.000000  50458.000000  37565.000000  46960.000000  50458.000000   
mean      -0.019809      0.864829     -0.003512     -0.020047      0.007739   
std        0.044403      0.061278      0.010821      0.038944      0.003717   
min       -0.222643      0.636588     -0.079389     -0.168573      0.002448   
25%       -0.051629      0.825744     -0.010255     -0.048239      0.004447   
50%       -0.022158      0.857518     -0.004011     -0.022548      0.007283   
75%        0.008066      0.896905      0.003049      0.004354      0.010421   
max        0.260459      1.497011      0.055841      0.215035      0.026108   

       sd_grd_eb_se  sd_mth_eb_se  sd_avg_ol_se  sd_grd_ol_se  sd_mth_ol_se  
count  37565.000000  46960.000000  50458.000000  37565.000000  46960.000000  
mean       0.004317      0.012381      0.008350      0.004867      0.013663  
std        0.001422      0.004771      0.004879      0.001889      0.005992  
min        0.001912      0.004819      0.002450      0.001940      0.004869  
25%        0.002956      0.007802      0.004493      0.003076  

In [100]:
desc_poolsub_cs

leaidC          fips       gap_est  tot_asmts_ela  tot_asmts_mth  \
count  7.214700e+04  72147.000000  72147.000000   7.204200e+04   7.184800e+04   
mean   2.932015e+06     29.187506      0.257585   9.754440e+03   8.960421e+03   
std    1.520158e+06     15.201987      0.437307   3.322655e+04   3.008000e+04   
min    1.000020e+05      1.000000      0.000000   2.000000e+01   2.000000e+01   
25%    1.734045e+06     17.000000      0.000000   1.243000e+03   1.173000e+03   
50%    2.926850e+06     29.000000      0.000000   3.245000e+03   3.061000e+03   
75%    4.108770e+06     41.000000      1.000000   8.347000e+03   7.846250e+03   
max    5.606240e+06     56.000000      1.000000   2.331930e+06   2.353281e+06   

       cellcount_ela  cellcount_mth  mn_asmts_ela  mn_asmts_mth  \
count   72042.000000   71848.000000  72042.000000  71848.000000   
mean       32.374879      30.630693    269.390770    269.704605   
std        12.340548      12.272864    906.428482    910.487181   
min         1.000000       1.000000     20.000000     20.000000   
25%        29.000000      25.000000     41.000000     41.151516   
50%        36.000000      35.000000     92.857140     93.047623   
75%        42.000000      41.000000    236.472923    236.694332   
max        42.000000      42.000000  70664.547000  71311.547000   

       mn_avg_mth_ol  mn_avg_ela_ol  mn_grd_mth_ol  mn_grd_ela_ol  \
count   66342.000000   66431.000000   46268.000000   39706.000000   
mean        0.059146       0.041812      -0.008441      -0.005112   
std         0.379982       0.377123       0.056426       0.043583   
min        -1.855213      -5.159541      -0.350875      -0.324992   
25%        -0.159971      -0.224729      -0.044345      -0.032920   
50%         0.033562       0.014001      -0.008844      -0.004648   
75%         0.288862       0.286326       0.026452       0.023193   
max         1.968772       2.159821       0.312431       0.195485   

       mn_avg_mth_eb  mn_avg_ela_eb  mn_grd_mth_eb  mn_grd_ela_eb  \
count   66540.000000   66586.000000   46585.000000   39929.000000   
mean        0.058966       0.041122      -0.008556      -0.005266   
std         0.372265       0.370975       0.047500       0.035819   
min        -1.676959      -1.851148      -0.314171      -0.253082   
25%        -0.150619      -0.224407      -0.038295      -0.027976   
50%         0.028197       0.012438      -0.008429      -0.005061   
75%         0.283986       0.285372       0.020850       0.018064   
max         1.948141       1.905320       0.225116       0.145515   

       mn_avg_mth_eb_se  mn_avg_ela_eb_se  mn_grd_mth_eb_se  mn_grd_ela_eb_se  \
count      66540.000000      66586.000000      46585.000000      39929.000000   
mean           0.032353          0.031137          0.018063          0.015136   
std            0.015806          0.014853          0.003873          0.002451   
min            0.005857          0.005965          0.004775          0.005118   
25%            0.022331          0.021663          0.015292          0.013484   
50%            0.027569          0.026806          0.017619          0.015060   
75%            0.036257          0.034893          0.020716          0.016877   
max            0.134209          0.129593          0.028179          0.041197   

       mn_avg_mth_ol_se  mn_grd_mth_ol_se  mn_avg_ela_ol_se  mn_grd_ela_ol_se  \
count      66342.000000      46268.000000      66431.000000      39706.000000   
mean           0.039907          0.021972          0.037100          0.019291   
std            0.032141          0.005625          0.027989          0.003784   
min            0.006052          0.006359          0.006054          0.006579   
25%            0.023087          0.017610          0.022285          0.016276   
50%            0.029216          0.020903          0.028245          0.018917   
75%            0.041123          0.025632          0.038840          0.022212   
max            0.235657          0.036310          0.2219

# Trying to make sense of the data

## NAEP

In [107]:
desc_long_naep

leaidC           fips          grade           year  \
count  8.258160e+05  825816.000000  825816.000000  825816.000000   
mean   3.003463e+06      29.903256       5.439089    2011.844344   
std    1.457768e+06      14.573536       1.694571       1.959203   
min    1.000020e+05       1.000000       3.000000    2009.000000   
25%    1.809690e+06      18.000000       4.000000    2010.000000   
50%    2.930660e+06      29.000000       5.000000    2012.000000   
75%    4.106300e+06      41.000000       7.000000    2013.000000   
max    5.606240e+06      56.000000       8.000000    2015.000000   

          totgyb_all         mn_all         sd_all      mn_all_se  \
count  825814.000000  825814.000000  825814.000000  825814.000000   
mean      328.138101     245.368960      30.441846       3.896056   
std      1166.280140      24.817281       5.184859       1.943400   
min        20.000000     114.207150       1.565124       0.363533   
25%        55.000000     228.196378      26.897011       2.418443   
50%       119.000000     245.202560      30.546568       3.439700   
75%       277.000000     262.570257      33.708389       5.099976   
max     76800.000000     375.458440     147.766400      61.499256   

           sd_all_se    totgyb_asn        mn_asn        sd_asn     mn_asn_se  \
count  825814.000000  85523.000000  85523.000000  85523.000000  85523.000000   
mean        2.408928    122.689382    264.213928     34.154915      6.580929   
std         1.575808    483.938421     30.265167      6.139830      2.941361   
min         0.248818     20.000000     99.107666      6.260830      0.529085   
25%         1.045357     29.000000    243.636095     30.374425      4.555845   
50%         2.302719     47.000000    263.100400     34.027737      6.535615   
75%         3.374068     97.000000    283.998410     37.234781      8.154646   
max        44.424931  12214.000000    441.867650    148.027940     74.957466   

          sd_asn_se     totgyb_blk         mn_blk         sd_blk  \
count  85523.000000  183171.000000  183171.000000  183171.000000   
mean       2.920028     224.051460     228.758834      29.498467   
std        2.652629     719.344149      22.675072       4.759979   
min        0.388104      20.000000     129.180240       3.078496   
25%        1.009118      36.000000     213.230470      26.277265   
50%        1.852622      72.000000     229.219020      29.588207   
75%        4.200429     170.000000     244.799475      32.436369   
max       69.068367   22678.000000     314.946560      90.725815   

           mn_blk_se      sd_blk_se     totgyb_fem         mn_fem  \
count  183171.000000  183171.000000  684574.000000  684574.000000   
mean        4.592179       2.268489     188.532721     248.236700   
std         2.103669       1.726871     621.838991      24.501244   
min         0.513911       0.312495      20.000000     123.957090   
25%         2.881899       0.750063      39.000000     230.850810   
50%         4.330511       1.956334      74.000000     247.870575   
75%         6.143965       3.353889     159.000000     265.586045   
max        41.399170      37.110298   37799.000000     403.577090   

              sd_fem      mn_fem_se      sd_fem_se     totgyb_hsp  \
count  684574.000000  684574.000000  684574.000000  222124.000000   
mean       29.727471       4.546764       2.357524     252.636262   
std         4.981161       2.008899       1.750996    1028.612933   
min         3.804324       0.488723       0.262531      20.000000   
25%        26.303948       2.984426       0.658222      33.000000   
50%        29.795918       4.273438       2.253879      66.000000   
75%        32.748504       5.934157       3.520308     177.000000   
max       103.046170      59.726406      42.617218   38791.000000   

              mn_hsp         sd_hsp      mn_hsp_se      sd_hsp_se  \
count  222124.000000  222124.000000  222124.000000  222124.000000   
mean      233.040101      29.822799       4.694741       2.091

The NAEP score is derived from the [National Assessment of Educational Progress](https://nces.ed.gov/nationsreportcard/ "NAEP Nations Report Card"). The test is on a 0-500 scale, which we can see reflected in the means of all the schools *(mn_all)*.

In [111]:
print(f'Minimum AVERAGE score for all schools: {desc_long_naep.loc["min","mn_all"]}')
print(f'Maximum AVERAGE score for all schools: {desc_long_naep.loc["max","mn_all"]}')

Minimum AVERAGE score for all schools: 114.20715
Maximum AVERAGE score for all schools: 375.45844


The above shows that the AVERAGE maximum score for any school is 375.5. This isn't to say that the highest score that a student received on the test is that number. Rather, that school could've had someone score the full 500 and another student who scored in the double digits. This is important to keep in mind while parsing through the data.

The other measures use various statistical techniques (HETOP and HOMOP, for example) to derive their normalized values, which I did not want to dig much deeper into. I believe using the NAEP score should be sufficient for the purposes of this data analysis.

Another interesting story comes from the "gaps" (disparity in average scores between sub-groups within a year/class/subject)

In [112]:
# Columns which contain values for the gaps between scores of:
# White v Asian students
#       v Black students
#       v Hispanic students
# Male v Female students

gap_cols= ["mn_wag","mn_wbg","mn_whg","mn_mfg"]

desc_long_naep.loc[:,gap_cols]

mn_wag         mn_wbg         mn_whg         mn_mfg
count  84024.000000  164413.000000  196243.000000  658277.000000
mean      -7.432532      20.692708      16.047852      -4.591977
std       14.264807       9.994392      10.275893       8.664576
min     -131.135770     -25.843262     -43.132675    -106.446230
25%      -16.002594      14.246857       9.267326      -9.915894
50%       -8.062675      20.054749      15.276093      -4.178528
75%        0.369026      26.478058      22.158958       1.042725
max      137.154270     147.544370     153.679550      58.579193

Looking at the min and max from the *mn_wag* column, it looks like there is a vast disparity between the scores of White students and Asian students (min: -131.1, max: 137.2). In fact, this suggests the opposite.

These scores are derived by subtracting the first subgroup ("W") from the second ("A"). So it appears that in at least one school, Asian student scores are __on average higher__ than their White peers by 131 points. The converse is true given the max value as well.

Armed with this information, we can see that there does indeed appear to be a disparity between White and Black students' as well as White and Hispanic students' scores. The same holds for Male v Female scores, with the disparity favoring Female students.